### PR TITLE
Relocate Rex::Platform:Windows content (fixes MS-1714)

### PR DESCRIPTION
**Pull request by pbarry-r7**
This PR consolidates the few lines of consts/code in lib/rex/platforms/windows.rb into MSF core.
## Verification

Load one of the modules which uses consts/code that this PR relocates and make sure it can still find what it needs (and doesn't throw an exception):
- [ ] Start `msfconsole`
- [ ] `use auxiliary/admin/backupexec/registry`
- [ ] `set RHOST <IP ADDR>` (doesn't actually need to be an IP for a responsive system, can be an usused/unresponsive IP)
- [ ] **Verify** the system sits for ~10 seconds and then prints out a failure due to "connection timed out"
- [ ] **Verify** you do **not** see a "NameError uninitialized constant" error for the REG_SZ const

### CODE REVIEW

File: lib/msf/core/post/windows/registry.rb [hunk 1]
```diff
@@ -30,6 +30,51 @@ module Registry
   REGISTRY_VIEW_64_BIT = 2
 
   #
+  # Windows Registry Constants.
+  #
+  REG_NONE = 1
```
Comment (pbarry-r7) [inline lib/msf/core/post/windows/registry.rb]: Is this a bug?  Should it be:

REG_NONE = 0

**Comment by pbarry-r7** [comment]
FWIW, I did see that lib/rex/post/meterpreter/extensions/stdapi/constants.rb
also these consts are defined, unclear to me if these are fine to leave as-is or if we should do a bit more coalescing...

**Comment by kgray-r7** [comment]
Looks good, merged.

**Comment by pbarry-r7** [comment]
ty, @kgray-r7!

**Comment by pbarry-r7** [comment]
# Release Notes

This is just a minor internal code reorg, no release notes needed.

**Pull request merged**